graphql-server/go-server/minio/service.go

```diff
@@ -0,0 +1,530 @@
+package minio
+
+import (
+	"context"
+	"encoding/json"
+	"encoding/xml"
+	"k8s.io/klog/v2"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	gouuid "github.com/satori/go.uuid"
+
+	models "github.com/kubeagi/arcadia/graphql-server/go-server/minio/model"
+)
+
+type SuccessChunksResult struct {
+	ResultCode int    `json:"resultCode"`
+	Uuid       string `json:"uuid"`
+	Uploaded   string `json:"uploaded"`
+	UploadID   string `json:"uploadID"`
+	Chunks     string `json:"chunks"`
+}
+
+type NewMultipartResult struct {
+	Uuid     string `json:"uuid"`
+	UploadID string `json:"uploadID"`
+}
+
+type MultipartUploadUrlResult struct {
+	Url string `json:"url"`
+}
+
+// completeMultipartUpload container for completing multipart upload.
+type CompleteMultipartUpload struct {
+	XMLName xml.Name             `xml:"http://s3.amazonaws.com/doc/2006-03-01/ CompleteMultipartUpload" json:"-"`
+	Parts   []minio.CompletePart `xml:"Part"`
+}
+
+// completedParts is a collection of parts sortable by their part numbers.
+// used for sorting the uploaded parts before completing the multipart request.
+type completedParts []minio.CompletePart
+
+func (a completedParts) Len() int           { return len(a) }
+func (a completedParts) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a completedParts) Less(i, j int) bool { return a[i].PartNumber < a[j].PartNumber }
+
+const (
+	PresignedUploadPartUrlExpireTime = time.Hour * 24 * 7
+)
+
+func GetSuccessChunks(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	var res = -1
+	var uuid, uploaded, uploadID, chunks, fileName string
+	var partNumberMarker, maxParts int
+
+	query := req.URL.Query()
+	fileMD5 := query.Get("md5")
+	if fileMD5 == "" {
+		klog.Error("GetFileChunkByMD5 failed: md5 is required")
+		_, err := w.Write([]byte("md5 is required"))
+		if err != nil {
+			klog.Errorf("w.Write failed: %s", err)
+		}
+		return
+	}
+
+	for {
+		fileChunk, err := models.GetFileChunkByMD5(fileMD5)
+		if err != nil {
+			klog.Infof("GetFileChunkByMD5 failed: %s", err)
+			break
+		}
+		uuid = fileChunk.UUID
+		uploaded = strconv.Itoa(fileChunk.IsUploaded)
+		uploadID = fileChunk.UploadID
+		fileName = fileChunk.FileName
+
+		bucketName := minioBucket
```

Comment:
For versioneddataset, the bucketname should be namespace. Not hardcoded

Reply:
TODO